aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -44,24 +45,31 @@ SparseCsrTensorImpl::SparseCsrTensorImpl(
               at::initialTensorOptions()
                   .device(SparseCsrTensorSetToDeviceType(key_set))
                   .dtype(data_type)) // values
+          ,
+          layout
       ) {}
 
 SparseCsrTensorImpl::SparseCsrTensorImpl(
     at::DispatchKeySet key_set,
     const caffe2::TypeMeta data_type,
     at::Tensor crow_indices,
     at::Tensor col_indices,
-    at::Tensor values)
+    at::Tensor values,
+    Layout layout)
     : TensorImpl(key_set, data_type, values.device()),
       crow_indices_(std::move(crow_indices)),
       col_indices_(std::move(col_indices)),
-      values_(std::move(values)) {
+      values_(std::move(values)),
+      layout_(layout) {
   set_storage_access_should_throw();
 }
 
 void SparseCsrTensorImpl::resize_(int64_t nnz, IntArrayRef size) {
-  auto rows = size[0];
-  auto cols = size[1];
+  TORCH_CHECK((layout_ == kSparseCsr || layout_ == kSparseBsr), "resize_: layout ", layout_, " is not yet supported");  // TODO
```

Comment:
With the assert in constructor you don't have to repeatedly test that the layout makes sense in these implementations

Reply:
We want to support the kSparseCsc and kSparseBsc as soon as possible. This check is to ensure that we don't forget to update this method because the code that follows would be wrong for CSC and BSC.